### PR TITLE
Add recipe for parquet-cpp

### PR DESCRIPTION
**Pull request by wesm**
This is unreleased software, but while it is in development, having packages in conda-forge will facilitate testing and developer engagement.

### CODE REVIEW

**Comment by wesm** [comment]
@jakirkham there's no perl in the linux-anvil. Can you recommend a workaround?

**Comment by wesm** [comment]
Might be a good idea to add perl to the docker container if possible.

**Comment by mariusvniekerk** [comment]
So you need perl in your test dependencies probably

**Comment by ocefpaf** [comment]
> Might be a good idea to add perl to the docker container if possible.

We had it before but decided to removed to avoid assuming a dependency that should be explicitly defined. If `perl` is needed only for the test @mariusvniekerk should work. 

Note also that conda defines the `CONDA_PERL` to a specific version, so it is safer to install the proper perl version and defined re-define the var explicitly to avoid issue with packages that have perl as a build/run dependency.

**Comment by jakirkham** [comment]
In addition to adding `perl`, you need to run `conda smithy rerender` locally and commit all changes. To get `conda-smithy`, simply install it from `conda-forge` into your `root` environment.

**Comment by wesm** [comment]
perl is only used as a minor utility in the script that runs the unit tests.  I'll try what you all have suggested and let you know if I have any problems

**Comment by wesm** [comment]
@jakirkham I'm afraid I don't have the attention span to figure out the conda smithy rerendering -- tried a bunch of things (e.g. `conda smithy rerender`, `conda smithy rerender recipes/parquet-cpp`, etc.) and got nowhere. Can you provide more explicit instructions?

**Comment by wesm** [comment]
Also @mariusvniekerk should I be putting that in circle.yml or meta.yml?

**Comment by mariusvniekerk** [comment]
Meta.yaml should be all you need

**Comment by jakirkham** [comment]
> @jakirkham I'm afraid I don't have the attention span to figure out the conda smithy rerendering -- tried a bunch of things (e.g. `conda smithy rerender`, `conda smithy rerender recipes/parquet-cpp`, etc.) and got nowhere. Can you provide more explicit instructions? 

Sorry please ignore what I said earlier, @wesm. That would only be necessary for the feedstock and will be done for you in this case.

**Comment by jakirkham** [comment]
I made a minor edit to your comment @mariusvniekerk to fix the yaml. Hope that is ok.

**Comment by wesm** [comment]
OK great. Let me try that and report back

**Comment by wesm** [comment]
Actually -- the issue is that the unit tests are being run in build.sh

https://github.com/conda-forge/staged-recipes/pull/992/files#diff-a411cd3149ae60abb902f6ec9ba9fdcaR44

As you can see I currently have perl in the build requirements, but it's not being picked up in the environment when the `run-test.sh` build support script is being run. This might be an issue of that script just not inheriting the conda bin/ path

**Comment by jakirkham** [comment]
That's fine. Just add `perl` to `requirements/build` instead.

**Comment by wesm** [comment]
I already have a failed build with it configured that way

**Comment by jakirkham** [comment]
Sorry, I missed that.

**Comment by jakirkham** [comment]
What's up with this error `find: `/llvm-3.7.0/bin': No such file or directory` in all of the tests?

**Comment by jakirkham** [comment]
So, the issue with `perl` is it seems that `parquet-cpp` expects to find `perl` in `/usr/bin/perl`. In reality it could be anywhere. My recommendation would be to submit a patch upstream that changes all lines like this `#!/usr/bin/perl` to `#!/usr/bin/env perl`. This way it will not depend on finding `perl` in a specific location and will grab the first one in `$PATH`.

**Comment by wesm** [comment]
Got it. Will do

**Comment by jakirkham** [comment]
Please feel free to include the equivalent patch here too until those changes work there way into a release.

**Comment by wesm** [comment]
Figured it was something silly so sorry about the noise. This is building from git master (no released versions yet) so it will be easy to patch

File: recipes/parquet-cpp/build.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+set -e
+set -x
+
+# FIXME: This is a hack to make sure the environment is activated.
+# The reason this is required is due to the conda-build issue
+# mentioned below.
+#
+# https://github.com/conda/conda-build/issues/910
+#
+source activate "${CONDA_DEFAULT_ENV}"
```
Comment (jakirkham) [inline recipes/parquet-cpp/build.sh]: This can be dropped as it does automatically.

File: recipes/parquet-cpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+source:
+  git_url: https://github.com/apache/parquet-cpp.git
+  git_rev: master
+
+build:
+  number: 0
+  script_env:
+    - CC [linux]
+    - CXX [linux]
+    - LD_LIBRARY_PATH [linux]
```
Comment (jakirkham) [inline recipes/parquet-cpp/meta.yaml]: Do we need these?

Comment (wesm) [inline recipes/parquet-cpp/meta.yaml]: Don't think so. I'm going to fix this patch up soon and get it building

**Comment by wesm** [comment]
@jakirkham this is good to go. library does not build on windows right now (see PARQUET-679, just needs a champion) so we can return and fix that later

File: recipes/parquet-cpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package:
+  name: parquet-cpp
+  version: "0.0"
+
+source:
+  git_url: https://github.com/apache/parquet-cpp.git
+  git_rev: master
```
Comment (ocefpaf) [inline recipes/parquet-cpp/meta.yaml]: @wesm pointing a recipe to `master` will render irreproducible builds. Can you change the top of your recipe to:

```yaml
{% set version = "0.0" %}
{% set commit = "c2e5f08cbefc3326535ef0ee99812b5123754d37" %}

package:
    name: parquet-cpp
    version: {{ version }}.dev0

source:
    fn: {{ commit }}.tar.gz
    url: https://github.com/apache/parquet-cpp/archive/{{ commit }}.tar.gz
    sha256: 1de1e8c2baf086d6fdffca6300d20166b5b9dbca082a0eae30a3c5b70d3e0cf4
```

I used the latest commit there BTW.

**Comment by chrisaycock** [comment]
Most of this LGTM. I assume LZ4 is no longer needed? Their [README](https://github.com/apache/parquet-cpp/blob/master/README.md) lists it as a dependency, but I noticed that it isn't checked for during the build. (Issue filed [here](https://issues.apache.org/jira/browse/PARQUET-744).)

**Comment by wesm** [comment]
Thank you

**Pull request merged**